mmv1/third_party/terraform/fwprovider/framework_provider_credentials_test.go

```diff
@@ -0,0 +1,283 @@
+					for _, v := range envvar.CredsEnvVars {
+						t.Setenv(v, badCreds)
+					}
+				},
+				Config:      testAccFwProvider_credentialsInEnvsOnly(context),
+				ExpectError: regexp.MustCompile("private key should be a PEM or plain PKCS1 or PKCS8"),
+			},
+			{
+				// GOOGLE_CREDENTIALS is used 1st if set
+				PreConfig: func() {
+					// good
+					t.Setenv("GOOGLE_CREDENTIALS", goodCredentials) //used
+					// bad
+					t.Setenv("GOOGLE_CLOUD_KEYFILE_JSON", badCreds)
+					t.Setenv("GCLOUD_KEYFILE_JSON", badCreds)
+					t.Setenv("GOOGLE_APPLICATION_CREDENTIALS", badCreds)
+				},
+				Config: testAccFwProvider_credentialsInEnvsOnly(context),
+			},
+			{
+				// GOOGLE_CLOUD_KEYFILE_JSON is used 2nd
+				PreConfig: func() {
+					// unset
+					t.Setenv("GOOGLE_CREDENTIALS", "")
+					// good
+					t.Setenv("GOOGLE_CLOUD_KEYFILE_JSON", goodCredentials) //used
+					// bad
+					t.Setenv("GCLOUD_KEYFILE_JSON", badCreds)
+					t.Setenv("GOOGLE_APPLICATION_CREDENTIALS", badCreds)
+
+				},
+				Config: testAccFwProvider_credentialsInEnvsOnly(context),
+			},
+			{
+				// GOOGLE_CLOUD_KEYFILE_JSON is used 3rd
+				PreConfig: func() {
+					// unset
+					t.Setenv("GOOGLE_CREDENTIALS", "")
+					t.Setenv("GOOGLE_CLOUD_KEYFILE_JSON", "")
+					// good
+					t.Setenv("GCLOUD_KEYFILE_JSON", goodCredentials) //used
+					// bad
+					t.Setenv("GOOGLE_APPLICATION_CREDENTIALS", badCreds)
+				},
+				Config: testAccFwProvider_credentialsInEnvsOnly(context),
+			},
+			{
+				// GOOGLE_APPLICATION_CREDENTIALS is used 4th
+				PreConfig: func() {
+					// unset
+					t.Setenv("GOOGLE_CREDENTIALS", "")
+					t.Setenv("GOOGLE_CLOUD_KEYFILE_JSON", "")
+					t.Setenv("GCLOUD_KEYFILE_JSON", "")
+					// good
+					t.Setenv("GOOGLE_APPLICATION_CREDENTIALS", goodCredentials) //used
+				},
+				Config: testAccFwProvider_credentialsInEnvsOnly(context),
+			},
+		},
+	})
+}
+
+func testAccFwProvider_credentials_emptyStringValidation(t *testing.T) {
+	acctest.SkipIfVcr(t) // Test doesn't interact with API
+
+	credentials := envvar.GetTestCredsFromEnv()
+
+	// ensure all credentials env vars set
+	for _, v := range envvar.CredsEnvVars {
+		t.Setenv(v, credentials)
+	}
+
+	context := map[string]interface{}{
+		"credentials":   "", // empty string used
+		"resource_name": "tf-test-" + acctest.RandString(t, 10),
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		// No PreCheck for checking ENVs
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccFwProvider_credentialsInProviderBlock(context),
+				PlanOnly:    true,
+				ExpectError: regexp.MustCompile("expected a non-empty string"),
+			},
+		},
+	})
+}
+
+// testAccFwProvider_credentialsInProviderBlock_provisionSdkResource is used when we want to make a plan to test
+// the credentials JSON path is valid (ie file exists) but we want to avoid a test failure cause by trying to use
+// those credentials, as they're fake.
```

Comment:
If we do a PlanOnly test step with the config using google_client_config then we get an error because the credentials are actively used to generate an access token, whereas a plan using a resource like google_service_account doesn't active try to use the credentials until apply time.

The alternative to this would be to migrate a data source (this is more conventional than google_client_config) back to the plugin-framework to be used here in this test. However we know that would be a breaking change until the muxing fixes are finished.

This currently doesn't test the PF provider but my intention is to update this to use a PF-implemented data source in the future.